tiledb/sm/serialization/array_schema.cc

```diff
@@ -160,12 +182,26 @@ static tuple<Status, optional<shared_ptr<Filter>>> filter_constructor(
     case FilterType::FILTER_BZIP2:
     case FilterType::FILTER_DOUBLE_DELTA:
     case FilterType::FILTER_DICTIONARY: {
-      auto data = reader.getData();
+      auto data = filter_reader.getData();
       int32_t level = data.getInt32();
       return {
           Status::Ok(),
           tiledb::common::make_shared<CompressionFilter>(HERE(), type, level)};
     }
+    case FilterType::FILTER_SCALE_FLOAT: {
+      if (!filter_reader.hasFloatScaleConfig()) {
+        throw std::logic_error(
```

Comment:
This `if` should be changed to the classic `if (filter_reader.hasFloatScaleConfig()) { call_all_getters(); }` approach. The reasoning is that `if(!reader.hasOption()) { throw exception(); }` would make all todays clients crash if in 10yrs from now we decide to remove this field from the CAPNP spec, the former classic approach protects us from this scenario, so it offers us more future flexibility for free.

Reply:
so the code should look like (pseudocode):
if (has float scale config) {
do code stuffs, return shared ptr
}

but like what happens when there is a failure in serialization? if we add a break and the error propogates down then the error message is wrong. should i change it so that it becomes, this would at least mitigate the error while acknowledging the correct type of the filter.

if (has float scale config) {
do code stuffs, return shared ptr
} else { return default filter }